openlibrary/plugins/admin/code.py

```diff
@@ -56,25 +56,25 @@ def delegate(self):
         for t in admin_tasks:
             m = web.re_compile('^' + t.path + '$').match(web.ctx.path)
             if m:
-                return self.handle(t.cls, m.groups())
+                return self.handle(t.cls, m.groups(), librarians=t.librarians)
         raise web.notfound()
 
-    def handle(self, cls, args=()):
+    def handle(self, cls, args=(), librarians=False):
         # Use admin theme
         context.bodyid = "admin"
 
         m = getattr(cls(), web.ctx.method, None)
         if not m:
             raise web.nomethod(cls=cls)
         else:
-            if self.is_admin():
+            if self.is_admin() or (librarians and context.user and context.user.is_librarian()):
                 return m(*args)
             else:
                 return render.permission_denied(web.ctx.path, "Permission denied.")
 
     GET = POST = delegate
 
-    def is_admin(self):
+    def is_admin(self, librarians=False):
```

Comment:
What is this `librarians` for? It should be in the doc string, but it doesn't appear to be used in the lines following?

Reply:
I don't think it needs to be here.